In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
sample = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

n = len(sample)

sample_array = np.array(sample)
mean = sample_array.mean()
s_std = sample_array.std()
print("\n Mean value               :",mean)
print("\n Standard deviation value :",s_std,"\n")

alpha = 0.01
t_crit = stats.t.ppf(1 - alpha/2, df=n-1)

margin = t_crit * (s_std / np.sqrt(n))

ci_lower = mean - margin
ci_upper = mean + margin

((ci_lower, ci_upper))



 Mean value               : 1.2386666666666666

 Standard deviation value : 0.18661427836285438 



(np.float64(1.095231668626123), np.float64(1.3821016647072102))

**Final interpretation :**

Based on the sample of 15 print-heads, the average durability is about 1.24 million characters, with a standard deviation of 0.19 million characters. Using the t-distribution, the 99% confidence interval for the mean durability is 1.10 to 1.38 million characters.

In [ ]:
p_std = 0.2
alpha = 0.01

z_crit = stats.norm.ppf(1 - alpha/2)

margin = z_crit * (p_std / np.sqrt(n))

ci_lower = mean - margin
ci_upper = mean + margin

(mean, (ci_lower, ci_upper))

(np.float64(1.2386666666666666),
 (np.float64(1.1056514133957607), np.float64(1.3716819199375725)))

**Final interpretation :**

Based on the sample of 15 print-heads, the average durability is about 1.24 million characters. Assuming a known population standard deviation of 0.20 million characters and using the z-distribution, the 99% confidence interval for the mean durability is 1.14 to 1.34 million characters.